### PR TITLE
Read vault token from ~/.vault-token if none supplied

### LINKED ISSUE #11365: vault provider should use `~/.vault-token` if `VAULT_TOKEN` isn't set
This allows simple operation of terraform if the user is already authenticated with vault.

### PR DESCRIPTION
**Pull request by eraserhd**
This closes #11365 (a feature request).

### CODE REVIEW

File: builtin/providers/vault/provider.go [hunk 1]
```diff
@@ -122,6 +124,18 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		return nil, fmt.Errorf("failed to configure Vault API: %s", err)
 	}
 
+	token := d.Get("token").(string)
+	if token == "" {
+		// Use the vault CLI's token, if present.
+		tokenFile := fmt.Sprintf("%s/.vault-token", os.Getenv("HOME"))
```
Comment (apparentlymart) [inline builtin/providers/vault/provider.go]: It looks like [vault's own implementation of this](https://github.com/hashicorp/vault/blob/f8c657a80a6b8e6dc5baa06febe0b67f00ced96e/command/token/helper_internal.go#L21) uses the `go-homedir` library, which tries various different means to locate the user's home directory, including some specialized support for Windows systems.

Do you think using that here would make sense, so we can ensure that the behavior will be consistent with Vault itself?

Comment (eraserhd) [inline builtin/providers/vault/provider.go]: Haha, I noticed and pushed a patch apparently just as you were commenting.  This is done.

**Comment by eraserhd** [comment]
I think the CI issue is not related.  This should be good to go.

File: builtin/providers/vault/provider.go [hunk 2]
```diff
@@ -122,6 +124,21 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		return nil, fmt.Errorf("failed to configure Vault API: %s", err)
 	}
 
+	token := d.Get("token").(string)
+	if token == "" {
+		// Use the vault CLI's token, if present.
+		homePath, err := homedir.Dir()
+		if err != nil {
+			return nil, fmt.Errorf("No vault token found: %s", err)
```
Comment (cblecker) [inline builtin/providers/vault/provider.go]: Wouldn't this error be called if the homedir was unable to be found? The error text might need to be clarified.

Comment (eraserhd) [inline builtin/providers/vault/provider.go]: Clarified.

**Comment by apparentlymart** [comment]
Thanks @eraserhd! This is merged in f1803ba4b6c5a5d6388d9977429f1637f6a251f5.

**Pull request closed without merging**